include/AST/Types/ArrayTy.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef GLU_AST_TYPES_ARRAYTY_HPP_
+#define GLU_AST_TYPES_ARRAYTY_HPP_
+
+#include "Types/TypeBase.hpp"
+#include <optional>
+
+namespace glu::types {
+
+/// @brief Represents an array type.
+class ArrayTy : public TypeBase {
+    TypeKind const _dataKind;
+    enum ArrayMode { Dynamic, Static } _mode;
+    std::optional<std::size_t> _size;
```

Comment:
You can add in documentation why _size is an optionnal thing and that _size is set just when the _mode is Static.